src/rust/engine/graph/src/lib.rs

```diff
@@ -444,7 +481,10 @@ impl<N: Node> Graph<N> {
       sleep(Duration::from_millis(500)).await;
 
       if let Some(inner) = Weak::upgrade(&inner) {
-        inner.lock().terminate_cycles();
+        let mut lock = inner.lock();
+        lock.terminate_cycles();
+        // TODO: figure out frequency to run this at
+        lock.prune_unreacheable_from_relevant();
```

Comment:
Probably the easiest way to run it at a different frequency would be to make a copy of this method, and spawn a separate task to execute at that frequency.